### PR TITLE
Newfood: Plates

### PR DESCRIPTION
**Pull request by Tsar-Salat**
## About The Pull Request

Completely replaces the existing plate code with a new system.

Instead of plates being attached to food sprites directly, the instead are dispensed from the Kitchen vendor in three sizes: Small, Medium, and Large.

These plates use a new shattering component, where they will break if thrown.

In addition, anticipating the increased usage of Serving Trays by Cooks, they can now be attached to belts.

Ports:
- https://github.com/tgstation/tgstation/pull/54224 (unit test)
- https://github.com/tgstation/tgstation/pull/57770
- https://github.com/tgstation/tgstation/pull/75303 (element only)

RESPRITES (the old ones hurt my eyes):
- https://github.com/tgstation/tgstation/pull/59171
- https://github.com/tgstation/tgstation/pull/75344

## Why It's Good For The Game

Necessary for ovens.

Also:

image
image

image
image

Old images were atrocious

## Testing Photographs and Procedure

<details>
<summary>Screenshots&Videos</summary>

Screenshot 2024-04-03 132737

</details>

## Changelog
:cl: rkz, PaxilManiac, Kryson, Qustinnus, nightred, Greniza, junkgle01
add: Separates plates from food sprites, into their own item. They come in three sizes, and break when thrown!
tweak: Serving Trays can now be attached to belts, anticipating its increased usage.
rscadd: resprites brioche, Vanilla, and ChocolateStrawberry cakes
code: removed duplicate update of plane, when layer is already changed
/:cl:

### CODE REVIEW

**Comment by Tsar-Salat** [comment]
i dont even think i fixed a quarter of the plates to not have the inbuilt plate sprites yet

gruh

**Comment by Tsar-Salat** [comment]
forgot about this

**Comment by Tsar-Salat** [comment]
<3

File: code/_onclick/hud/radial.dm
Comment (PowerfulBacon) [inline code/_onclick/hud/radial.dm:1]: These changes seem completely unrelated

File: code/_onclick/click.dm
Comment (PowerfulBacon) [inline code/_onclick/click.dm:1]: These changes seem unrelated, what are they for?

File: code/datums/components/storage/storage.dm
Comment (PowerfulBacon) [inline code/datums/components/storage/storage.dm:1]: These changes seem unrelated, what are they for?

**Review by PowerfulBacon — COMMENTED** [review]
The code seems good though

File: code/_onclick/click.dm
Comment (Tsar-Salat) [reply code/_onclick/click.dm:1]: The plates both contain and hold the food, but in way that you can eat them and not the same storage bag way food trays do

File: code/_onclick/hud/radial.dm
Comment (Tsar-Salat) [reply code/_onclick/hud/radial.dm:1]: As per the original pr, items in inventory should only be changing a singular variable. That they are changing both planes and layers which is weird.

I neglected to put this in the CL, which I have now rectified. Thank you for the review

File: code/datums/components/storage/storage.dm
Comment (Tsar-Salat) [reply code/datums/components/storage/storage.dm:1]: Defines which are used outside of a file probably shouldn't be in that file alone.

File: code/_onclick/hud/radial.dm
Comment (PowerfulBacon) [reply code/_onclick/hud/radial.dm:1]: /atom/movable/screen/radial does define plane so it should be still on the same plane. Changing both plane and layers in 1 place makes perfect sense though, planes are fundamentally different to layers.

Layers define the z-ordering of atoms within a plane.
Planes define a 'draw surface' in which atoms are temporarilly drawn to before being drawn to the screen. Planes are sorted among themselves, but layer only applies locally to that plane.

A: Plane 2, Layer 1
B: Plane 1, Layer 1
C: Plane 2, Layer 2
D: Plane 1, Layer 2

The draw ordering would be: (C, A, D, B), with C and A belonging to plane 2 and D and B belonging to plane 1 (which may have special filters/transformations/effects applied to their draw surface.)

**Review by PowerfulBacon — APPROVED** [review]
Seems fine

File: code/_onclick/hud/radial.dm
Comment (Tsar-Salat) [reply code/_onclick/hud/radial.dm:1]: fine. [bfae66b](https://github.com/BeeStation/BeeStation-Hornet/pull/10920/commits/bfae66b4999e14013f9b81f3ae7da2b859ec39bb)

File: code/_onclick/hud/radial.dm [hunk 1]
```diff
@@ -242,7 +242,7 @@ GLOBAL_LIST_EMPTY(radial_menus)
 			var/datum/radial_menu_choice/choice_datum = choice_datums[choice_id]
 			if (choice_datum.info)
 				var/obj/effect/abstract/info/info_button = new(E, choice_datum.info)
-				info_button.plane = ABOVE_HUD_PLANE
+				nfo_button.plane = ABOVE_HUD_PLANE
```
Comment (PowerfulBacon) [inline code/_onclick/hud/radial.dm]:
```suggestion
				info_button.plane = ABOVE_HUD_PLANE
```

**Comment by Rukofamicom** [comment]
Woo
